rust/agents/scraper/src/agent.rs

```diff
@@ -88,12 +91,19 @@ impl BaseAgent for Scraper {
             core_metrics: metrics,
             agent_metrics,
             chain_metrics,
+            server,
```

Comment:
nit: Storing the server in the agent structs shouldn't be required, since the core_metrics are accessible from within `run()`. We can construct the server in there directly, for all agents